File: web/src/main/java/top/bettercode/summer/web/filter/DefaultApiVersionService.java
package top.bettercode.summer.web.filter;

import top.bettercode.summer.web.config.SummerWebProperties;

/**
 * @author Peter Wu
 */
public class DefaultApiVersionService implements ApiVersionService{

  private final SummerWebProperties summerWebProperties;

  public DefaultApiVersionService(SummerWebProperties summerWebProperties) {
    this.summerWebProperties = summerWebProperties;
  }

  @Override
  public String getVersionName() {
    return summerWebProperties.getVersionName();
  }

  @Override
  public String getVersion() {
    return summerWebProperties.getVersion();
  }

  @Override
  public String getVersionNoName() {
    return summerWebProperties.getVersionNoName();
  }

  @Override
  public String getVersionNo() {
    return summerWebProperties.getVersionNo();
  }

}

File: web/src/main/kotlin/top/bettercode/summer/web/filter/DefaultApiVersionService.kt
package top.bettercode.summer.web.filter

import top.bettercode.summer.web.properties.SummerWebProperties

/**
 * @author Peter Wu
 */
class DefaultApiVersionService(private val summerWebProperties: SummerWebProperties) : IApiVersionService {
    override val versionName: String?
        get() = summerWebProperties.versionName
    override val version: String?
        get() = summerWebProperties.version
    override val versionNoName: String?
        get() = summerWebProperties.versionNoName
    override val versionNo: String?
        get() = summerWebProperties.versionNo
}
